orchestra/todos/api.py

```diff
@@ -100,7 +100,8 @@ def _add_template_todo(
             template=todolist_template,
             parent_todo=parent_todo,
             status=status,
-            additional_data=additional_data
+            additional_data=additional_data,
+            required=template_todo.get('required', False)
```

Comment:
Don't overthink this---just starting a discussion that I don't have a strong opinion about: are we overloading concepts too much? We have to-do slugs and we have `required`, and in some cases (internal codebase) we rely on the slug to identify its qualities and others we rely on a specific property like `required`.
* Argument for implementing this PR: `required` is a pretty foundational concept, and stands alone.
* Argument for not implementing this PR: We already rely on the slug to identify required status in some internal functionality like action links---do we want to have two ways to make a to-do required?

Reply:
General cases: in certain cases we want to treat todos as required except for a small number of todos

- pros: we already have some logic which checks if required todos are completed, as in, required field is already being used

Special cases: in some cases we want to treat a smaller set of todos as required

- pros: since all todos except some edge cases are marked as required, the `required` field is not helpful here. Here we want to use `slug`

Reply:
My argument for adding it was the same as @marcua listed -- `required is a pretty foundational concept, and stands alone.`